lib/manager.js

```diff
@@ -65,6 +65,7 @@ function Manager (uri, opts) {
   var _parser = opts.parser || parser;
   this.encoder = new _parser.Encoder();
   this.decoder = new _parser.Decoder();
+  this.engineFactory = opts.engineFactory || defaultEngineFactory;
```

Comment:
Something to consider... You could avoid the need for the `defaultEngineFactory` function by changing this to:
```
opts.engineFactory || eio
```

Reply:
or at least call `new`

Reply:
going to leave as is, we'll worry about changes when I try to merge upstream.